src/components/pages/jobs/Job.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import styles from './job.module.scss';
+import { JobItem } from 'models/apiModels';
+import Icon from 'components/atoms/icons/icon';
+import { getJob, deleteJob } from 'api/jobs';
+import { baseUrl } from 'constants/apiConstants';
+import { AuthenticateContext, Roles } from 'contexts/authProvider';
+import Modal from 'components/molecules/modal/Modal';
+import { useMobileScreen } from 'hooks/useMobileScreen';
+import { useHistory } from 'react-router-dom';
+import { useToast } from 'hooks/useToast';
+import Button from 'components/atoms/button/Button';
+
+const Job = () => {
+  const isMobile = useMobileScreen();
+  const { id } = useParams<{ id: string }>();
+  const { addToast } = useToast();
+  const { role } = useContext(AuthenticateContext);
+  const history = useHistory();
+  const [showModal, setShowModal] = React.useState<boolean>(false);
+  const [data, setData] = React.useState<JobItem | undefined>(undefined);
+  const imgUrl = baseUrl + 'jobs/' + data?.id + '/image';
+  const [isValidJob, setIsValidJob] = React.useState<boolean>(false);
+  useEffect(() => {
+    const getData = async (id: string) => {
+      try {
+        const res: JobItem = await getJob(id);
+        setData(res);
+        setIsValidJob(true);
+      } catch (error) {
+        setIsValidJob(false);
+      }
+    };
+    getData(id);
+  }, [id]);
+  return (
+    <div>
+      {isValidJob ? (
```

Comment:
Is this state necessary? It seems like it's derived from if `data` is something. It also doesn't look like the loading will take effect since `isValidJob` will be checked first, and it will then show `404 job not found`